src/base/moduledownloader.lua

```diff
@@ -0,0 +1,125 @@
+--
+-- moduledownloader.lua
+-- Downloads a module from a package server
+-- Copyright (c) 2002-2016 Jason Perkins and the Premake project
+--
+
+
+---
+-- progress bar.
+--
+	local function _http_progress(total, current)
```

Comment:
works for me, the only thing I would add is that on build servers this is annoying as hell, since the \r doesn't work in captured console output, and so you just get hundreds of lines of this in your log... so a way to disable it would be nice, or have it disabled by default, and only enable with --verbose... either way...